lib/core/versions/latest/Did.ts

```diff
@@ -44,17 +46,17 @@ export default class Did {
     }
 
     if (this.isShortForm) {
-      this.uniqueSuffix = did.substring(didMethodName.length);
+      this.uniqueSuffix = did.substring(didPrefix.length);
     } else {
       // This is long-form.
-      this.uniqueSuffix = did.substring(didMethodName.length, indexOfQuestionMarkChar);
+      this.uniqueSuffix = did.substring(didPrefix.length, indexOfQuestionMarkChar);
```

Comment:
Couldn't these be combined with 43/45? It makes substring'ing on indexOfQuestionMarkChar >=0 at compile time.

Reply:
Not sure if the ask is to combine the two if statements. If so, I'd prefer to keep them separate for readability for dummies.